### PR TITLE
PanelModel supports RefreshMode.

### PR DESCRIPTION
**Pull request by TomTirapani**
Also, all models which implemented RefreshMode use the new RefreshModeContextModel.

See Issue: https://github.com/xh/hoist-react/issues/1310
See Toolbox PR: https://github.com/xh/toolbox/pull/382

Hoist P/R Checklist
-------------------

**Pull request authors:** Review and check off the below. Items that do not apply can also be
checked off to indicate they have been considered. If unclear if a step is relevant, please leave
unchecked and note in comments.

- [x] Caught up with `develop` branch as of last change.
- [x] Added CHANGELOG entry, or determined not required.
- [x] Reviewed for breaking changes, added `breaking-change` label + CHANGELOG if so.
- [x] Updated doc comments / prop-types, or determined not required.
- [x] Reviewed and tested on Mobile, or determined not required.
- [x] Created Toolbox branch / PR, or determined not required.

**If your change is still a WIP**, please use the "Create draft pull request" option in the split
button below to indicate it is not ready yet for a final review.

> **Pull request reviewers:** when merging this P/R, please consider using a *squash commit* to
> collapse multiple intermediate commits into a single commit representing the overall feature
> change. This helps keep the commit log clean and easy to scan across releases. PRs containing a
> single commit should be *rebased* when possible.

### CODE REVIEW

File: desktop/cmp/panel/PanelModel.js [hunk 1]
```diff
@@ -47,6 +51,13 @@ export class PanelModel {
     /** Is this panel currently resizing? */
     @observable isResizing = false;
 
+    /**
```
Comment (lbwexler) [inline desktop/cmp/panel/PanelModel.js]: no need for comment

File: desktop/cmp/panel/impl/ResizeContainer.js [hunk 2]
```diff
@@ -37,12 +38,15 @@ export const resizeContainer = hoistCmp.factory({
         const cmp = vertical ? vbox : hbox,
             maxDim = vertical ? 'maxHeight' : 'maxWidth';
 
-        return cmp({
-            ref,
-            className,
-            flex: 'none',
-            [maxDim]: '100%',
-            items
+        return refreshContextView({
```
Comment (lbwexler) [inline desktop/cmp/panel/impl/ResizeContainer.js]: My only concern here is that this refreshContext only exists for a resizable/collapsible panel. Maybe that is what we want, but we got burned by overly branching the structure between the two types of panel before.....

**Comment by lbwexler** [comment]
This is really awesome, and a nice DRY up.  A couple of thoughts:

1)  Would it be worth calling this a LazyRefreshContextModel?  Realize it is not always lazy, but that is the default, and is its main reason for existence.  Could also be called a ManagedRefreshContextModel or an InactiveContentRefreshContextModel.

2)

File: desktop/cmp/panel/impl/ResizeContainer.js (hunk 2 above)
Comment (TomTirapani) [reply desktop/cmp/panel/impl/ResizeContainer.js]: Good point - will move to the main panel component, would be less surprising

**Comment by TomTirapani** [comment]
Updated this branch with comments from the code review - thanks!

Will also add the Toolbox example shortly

**Comment by lbwexler** [comment]
Found some issues in this we need to work out.  Seems like getting some strange behavior with registration of models within this context.  For example, notice the refresh buttons in toolboxes toolbars no longer seem to work

**Comment by lbwexler** [comment]
Update -- not seeing any strange behavior -- that was a user error.  Have been discussing with Tom, however, and have the following thoughts:

1) Definitely not worth the complexity of all of the additional refresh context models to insert these for non-collapsible panels.

2) Might not even be worth it in the case of collapsible panels.  We need to discuss.

**Comment by lbwexler** [comment]
I would be inclined to merge this now.  I think its a nice cleanup/standardization.  We don't have a strong example yet of the use of lazy refresh in the collapsed panel, but it seems analogous to TabContainer and its nice to bring these all in to line.

**Comment by TomTirapani** [comment]
Great. I've got a Toolbox example in the Portfolio app cooking - will link here once available

**Pull request merged**